rust/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -144,7 +134,29 @@ impl BaseMetadataBuilder {
             .await
             .context(CTX)?;
         checkpoint_syncer
-            .fetch_checkpoint_in_range(validators, threshold, message.nonce, highest_known_nonce)
+            .legacy_fetch_checkpoint_in_range(
+                validators,
+                threshold,
+                message.nonce,
+                highest_known_nonce,
+            )
+            .await
+            .context(CTX)
+    }
+
+    pub async fn fetch_checkpoint(
```

Comment:
You need two different versions of this function for censorship resistance.

For merkle proof ISMs, you can use *any* checkpoint with nonce >= message nonce. You will need something similar to `fetch_checkpoint_in_range`

For message ID ISMs, you need the specific checkpoint that matches the message nonce, which is what you are doing here.